src/main/java/hudson/tasks/junit/History.java

```diff
@@ -551,24 +479,25 @@ private ObjectNode computeDistributionJson(List<HistoryTestResultSummary> histor
             mul = 1.0d / 3600.0d;
             roundMul = 100.0;
         }
+        xAxis.put("min", (float) (mul * lrX[0] - (step * mul * 0.5)));
+        xAxis.put("max", (float) (mul * lrX[lrX.length - 1] + (step * mul * 0.5)));
+        xAxis.put("interval", (float) (step * mul));
+        xAxis.put("roundingFactor", (float) roundMul);
 
         int maxBuilds = 0;
-        SmoothingCubicSpline scs = new SmoothingCubicSpline(lrX, lrY, 0.1);
-        int smoothPts = counts.length * 4;
-        double k = (double) counts.length / smoothPts;
-        final double splineRoundMul = 1000.0;
-        for (double z = minDuration; z < maxDuration; z += step * k) {
-            double v = Math.round(splineRoundMul * Math.max(0.0, scs.evaluate(z / scale * 100.0))) / splineRoundMul;
-            durationData.add((float) v);
-            maxBuilds = Math.max(maxBuilds, (int) Math.ceil(v));
-            // Use float for smaller JSONs.
-            domainAxisLabels.add((float) (Math.round(mul * z * roundMul) / roundMul));
+        for (int i = 0; i < lrY.length; i++) {
+            double scaledX = mul * lrX[i];
```

Comment:
We no longer round our X values in the backend (`z` in the old code), because our axis type is now `value` instead of `category`, and so if we round here our bars will have weird gaps and not be contiguous.